pyblish_nuke/lib.py

```diff
@@ -255,6 +256,37 @@ def __init__(self):
 def dock(window):
     """ Expecting a window to parent into a Nuke panel, that is dockable. """
 
+    # Deleting existing dock
+    # There is a bug where existing docks are kept in-memory when closed via UI
+    if self._dock:
+        print "Deleting existing dock..."
+        self._dock.layout().addWidget(window)
+
+        parent = self._dock
+        dialog = None
+        stacked_widget = None
+        main_windows = []
+        # getting dock parents
+        while parent:
+            if parent.__class__.__name__ == "QDialog":
```

Comment:
This looks like an ad-hoc alternative to `isinstance(parent, QtWidgets.QDialog)`, could we make it that instead?